presto-common/src/main/java/com/facebook/presto/common/type/LongEnumType.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.function.SqlFunctionProperties;
+
+import java.util.Map;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.TypeUtils.validateEnumEntries;
+
+public class LongEnumType
+        extends AbstractLongType
```

Comment:
Would this allow casting from other numeric types to `LongEnumType` ? eg. `CAST(1.01 AS Country)`. It will get tricky if there is implicit casting

Reply:
No, the Java class hierarchy of the type classes is separate from function signature resolution in Presto itself.

With the operators introduced in EnumOperators, I only allow casting from smallint/tinyint/integer/bigint to numeric enum.